common/src/main/java/tachyon/worker/WorkerClient.java

```diff
@@ -428,4 +433,21 @@ public synchronized void sessionHeartbeat() throws IOException {
       throw new IOException(e);
     }
   }
+
+  /**
+   * Called only by {@link WorkerClientHeartbeatExecutor}, encapsulates
+   * {@link #sessionHeartbeat()} in order to cancel and cleanup the
+   * heartbeating thread in case of failures
+   */
+  public synchronized void periodicHeartbeat() {
```

Comment:
I think the use of `sessionHeartbeat` in `BlockWorkerClientPool` would also benefit from this error handling, but perhaps we can refactor this later, since `BlockWorkerClientPool` doesn't really want to heartbeat but just send metrics.